cirq/experiments/random_quantum_circuit_generation.py

```diff
@@ -289,16 +294,65 @@ def _get_active_pairs(graph: nx.Graph, grid_layer: GridInteractionLayer):
 
 @dataclasses.dataclass(frozen=True)
 class CircuitLibraryCombination:
-    """For a given layer (which defines a set of pairs of qubits), `combinations` is a 2d array
+    """For a given layer (specifically, a set of pairs of qubits), `combinations` is a 2d array
     of shape (n_combinations, len(pairs)) where each row represents a combination (with replacement)
     of two-qubit circuits. The actual values are indices into a list of library circuits.
 
-    `layer` is used for record-keeping but is otherwise unnecessary. Therefore, it is optional.
+    `layer` is used for record-keeping. This is the GridInteractionLayer if using
+    `get_random_combinations_for_device`, the Moment if using
+    `get_random_combinations_for_layer_circuit` and ommitted if using
+    `get_random_combinations_for_pairs`.
     """
 
-    layer: Optional[GridInteractionLayer]
+    layer: Optional[Any]
```

Comment:
Instead of Any, should we list the types as a Union?  It is only three different possibilities, right?

Reply:
Yes, I thought about this. The strongest argument for `Any` is that someone might want to write a new wrapper for `_get_random_combinations` with a custom `pair_gen` iterable. I wanted to allow the second item yielded to be any metadata that might be useful for provenance of the particular layer. It should never actually be consumed by the parallel XEB functionality. In some sense, having this broad type will make sure no one relies on this field being one of the couple of types the present functionality may result in. 

I suppose it will always be backwards compatible to broaden the scope of a type by having it be a union now and a bigger union later as applicable, and an `Any` if it comes to that. What do you think given this additional context?

Reply:
gonna go ahead and merge now so I can open the next PR. Feel free to follow-up and we can make the change there or elsewhere if desired